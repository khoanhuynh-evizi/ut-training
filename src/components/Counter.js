import React, { useState } from 'react';

const Counter = () => {
  const [counter, setCounter] = useState(0);

  const decrement = () => setCounter(counter - 1)

  const reset = () => setCounter(0)

  const random = () => setCounter(Math.round(Math.random() * 100))

  const increment = () => setCounter(counter + 1)

  return (
    <div className='counter'>
      <h2>{counter}</h2>

      <button className="decrement" onClick={decrement}>-</button>
      <button className="reset" onClick={reset}>reset</button>
      <button className="random" onClick={random}>random</button>
      <button className="increment" onClick={increment}>+</button>

    </div>
  );
}

export default Counter