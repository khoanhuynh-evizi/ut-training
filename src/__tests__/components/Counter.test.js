import React from "react";
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Counter from '../../components/Counter'

let wrapper 

beforeEach(() => {
  wrapper = shallow(<Counter />);
  jest.spyOn(global.Math, 'random').mockReturnValue(0.2308);
});

afterEach(() => {
  jest.spyOn(global.Math, 'random').mockRestore();
})


it('should match the snapshot', () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

describe('renders without error', () => {
  it("renders correct", () => {
    expect(wrapper.find('.counter')).toHaveLength(1);
  });

  it("renders buttons", () => {
    expect(wrapper.find('.decrement')).toHaveLength(1);
    expect(wrapper.find('.reset')).toHaveLength(1);
    expect(wrapper.find('.increment')).toHaveLength(1);
  });
  
  it("renders counter is zero", () => {
    expect(wrapper.find('h2').text()).toBe("0")
  });
})

describe("Increment", () => {
  it("click increment button", () => {
    const wrapper = shallow(<Counter />);

    const increment = wrapper.find('.increment');
    increment.simulate('click');

    expect(wrapper.find('h2').text()).toBe("1")
  });
});


describe("Decrement", () => {
  it("click decrement button", () => {
    const wrapper = shallow(<Counter />);

    const decrement = wrapper.find('.decrement');
    decrement.simulate('click');

    expect(wrapper.find('h2').text()).toBe("-1")
  });
});


describe("Reset", () => {
  it("click reset button", () => {
    const wrapper = shallow(<Counter />);

    const decrement = wrapper.find('.decrement');
    decrement.simulate('click');

    expect(wrapper.find('h2').text()).toBe("-1")

    const reset = wrapper.find('.reset');
    reset.simulate('click');

    expect(wrapper.find('h2').text()).toBe("0")

  });
});


describe("Random", () => {
  it("click random button", () => {
    const wrapper = shallow(<Counter />);

    const decrement = wrapper.find('.decrement');
    decrement.simulate('click');

    expect(wrapper.find('h2').text()).toBe("-1")

    const reset = wrapper.find('.random');
    reset.simulate('click');

    expect(wrapper.find('h2').text()).toBe("23")

  });
});
