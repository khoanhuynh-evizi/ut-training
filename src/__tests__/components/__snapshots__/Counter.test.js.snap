// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="counter"
>
  <h2>
    0
  </h2>
  <button
    className="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <button
    className="reset"
    onClick={[Function]}
  >
    reset
  </button>
  <button
    className="random"
    onClick={[Function]}
  >
    random
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
